src/git.rs

```diff
@@ -11,8 +12,48 @@ pub fn git_fetch(root: &Path) -> Result<(), git2::Error> {
     Ok(())
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct BranchInfo {
+    name: Option<String>,
```

Comment:
I'm honestly surprised Git even allows this non-utf8 branch names.

I would be curious to see how GitHub behaves and what their APIs return on a non-utf8 branch name and/or message.

It might be worth ignoring these and simplifying the types and schema a bit.

Reply:
sounds good, ive done the lossy conversion